update e2e environment for new timestamp contracts

weilbith opened the pull request:
Fixes #3050

Note that all versions point to custom release-candiate git tags that I created on all the components repositories. Especially the Light Client itself points to the `arbitrum` branch. We will need a final update of the image, pointing to official releases.
Anyway I have allowed myself to push [a release-candidate](https://hub.docker.com/layers/raidennetwork/lightclient-e2e-environment/v2.0.0rc1/images/sha256-9ac4eb107a07f8250d6d5ab4cd9dc1a29b47b132241536e7e62f00a565cf0a9c?context=explore) of the new image for convenience to use for #3034 (note that this requires to touch `e2e-environment/shared-script.sh#8`).

weilbith commented:
I also pushed a pre-release version of the new Docker image to the hub and referenced it in the CircleCI configuration. Let's see if we can reproduce the success from my local machine here.

weilbith commented:
I'll now rebase on latest `arbitrum`. The result should be a single commit.

weilbith commented:
The `lint` job is still expected to fail due to issues in the upstream branch. This will not be fixed here. All other jobs should pass here.

weilbith commented:
The `lint` job is expected to fail right now. A fix for this would be totally unrelated to the changes here. Therefore we will do this in the `arbitrum` upstream branch.

netcriptus commented on e2e-environment/Dockerfile:
```diff
@@ -1,9 +1,9 @@
-ARG CONTRACTS_VERSION="0.40.0"
-ARG SERVICES_VERSION="v1.0.0"
+ARG RAIDEN_VERSION="arbitrum"
+ARG CONTRACTS_PACKAGE_VERSION="v1.0.0rc2"
```
There's an RC3 now, maybe it's a good idea to update

weilbith replied:
Is there? What does it include? The second release candidate includes all relevant changes that are not release procedure related, right?
Note that this will change anyway soon...

netcriptus replied:
It includes the version bump, but it's fine to move on with this one then.

weilbith replied:
✔️

netcriptus commented on e2e-environment/Dockerfile:
```diff
@@ -32,7 +35,8 @@ ARG CONTRACTS_VENV
 RUN python -m venv ${CONTRACTS_VENV}
-RUN . ${CONTRACTS_VENV}/bin/activate && pip install -U pip wheel && pip install raiden-contracts==${CONTRACTS_PACKAGE_VERSION}
+RUN ${CONTRACTS_VENV}/bin/pip install --upgrade pip wheel
+RUN ${CONTRACTS_VENV}/bin/pip install git+https://github.com/weilbith/raiden-contracts@${CONTRACTS_PACKAGE_VERSION}
```
Why are we getting it from your fork instead of the main repository?

weilbith replied:
At this point in time I did not want to spam the `origin` repos with such experimental tags. And as it will not stay like that anyway and will change, I thought that is fine.
Do you want me to change it? 🙃

netcriptus replied:
As an effect, we'll have an `origin` repo pointing to a fork, which I honestly believe is worse. I'd prefer to have this pointing to a branch, but on a main repo.

weilbith replied:
Sure. I'll do.

weilbith replied:
✔️

netcriptus reviewed (commented):
Looking very good, I just need some clarification for one comment, and we might want to consider the new RC

weilbith commented:
I addressed the review feedback. I also pointed to the new `raiden-contracts` tag after raiden-network/raiden-contracts#1614 . Finally I rebased on `arbitrum` which should make the `lint` CI job succeed too.

weilbith commented:
I now pushed the new image to DockerHub. This should be pulled by CircleCI and make the pipeline succeed.
